src/components/Switch/Switch.tsx

```diff
@@ -1,184 +1,231 @@
-import { forwardRef, memo, useMemo } from 'react';
-import type { ComponentPropsWithRef } from 'react';
-import {
-    NativeModules,
-    Platform,
-    StyleProp,
-    Switch as NativeSwitch,
-    ViewStyle,
-} from 'react-native';
-import setColor from 'color';
+import { forwardRef, memo, useCallback, useEffect, useMemo, useRef } from 'react';
+import { Animated, Easing, Pressable, SwitchProps, ViewStyle } from 'react-native';
+import { useActionState, useComponentStyles, useMolecules } from '../../hooks';
 
-import type { ComponentStylePropWithVariants } from '../../types';
-import { useComponentStyles } from '../../hooks';
+export type ToggleProps = SwitchProps & {
+    checkedIcon?: string;
+    unCheckedIcon?: string;
+    size?: number;
+    checkedIconType?: IconType;
+    uncheckedIconType?: IconType;
+};
 
-const version = NativeModules.PlatformConstants
-    ? NativeModules.PlatformConstants.reactNativeVersion
-    : undefined;
+type IconType =
+    | 'material'
+    | 'material-community'
+    | 'simple-line-icon'
+    | 'zocial'
+    | 'font-awesome'
+    | 'octicon'
+    | 'ionicon'
+    | 'feather'
+    | 'fontisto'
+    | 'foundation'
+    | 'evilicon'
+    | 'entypo'
+    | 'antdesign'
+    | 'font-awesome-5'
+    | undefined;
 
-export type Props = ComponentPropsWithRef<typeof NativeSwitch> & {
-    /**
-     * Disable toggling the switch.
-     */
-    disabled?: boolean;
-    /**
-     * Value of the switch, true means 'on', false means 'off'.
-     */
-    value?: boolean;
-    /**
-     * Custom color for switch.
-     */
-    color?: string;
-    /**
-     * Callback called with the new value when it changes.
-     */
-    onValueChange?: Function;
-    style?: StyleProp<ViewStyle>;
-};
+const Switch = ({
+    trackColor,
```

Comment:
trackColor and thumbColor should go through useComponentStyles in case they are tokens